src/main/java/ch/njol/skript/effects/EffActionBar.java

```diff
@@ -0,0 +1,76 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.effects;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.util.Kleenean;
+
+@Name("Action bar")
+@Description("Sends an action bar message to the given player.")
+@Examples("send action bar \"Action bar message.\" to player")
+@Since("INSERT VERSION")
+public class EffActionBar extends Effect {
+
+	static {
+		if (Skript.methodExists(Player.class, "sendActionBar", String.class)) {
+			Skript.registerEffect(EffActionBar.class, "[send] action bar %string% [to %players%]");
+		}
+	}
+
+	@SuppressWarnings("null")
+	private Expression<String> message;
+
+	@SuppressWarnings("null")
+	private Expression<Player> recipients;
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final SkriptParser.ParseResult parser) {
+		message = (Expression<String>) exprs[0];
+		recipients = (Expression<Player>) exprs[1];
+		return true;
+	}
+
+	@Override
+	protected void execute(final Event e) {
+		for (Player player : recipients.getArray(e)) {
+			for (String string : message.getArray(e)) {
+				player.sendActionBar(string);
```

Comment:
This is not a method in spigot. When a script attempts to run this effect on spigot, this will fail. You must test for paper in the init method if you want to do this effectively. However, this can be done in plain spigot itself so I do not think paper should be required for this effect.

Reply:
Wouldn't that never happen because they're not registering it unless the method exists? In theory none of the code in this class should ever run if the effect isn't registered.

Reply:
Yes, I didn't see that check in the static call, whoops! But still, we can add non-nms action bar using bungee's api so it will work on every spigot server.